src/include/firebird/FirebirdInterface.idl

```diff
@@ -1695,3 +1696,49 @@ interface ReplicatedSession : PluginBase
 	void setSequence(Status status, const string name, int64 value);
 }
 
+// Profiler interfaces
+
+interface ProfilerPlugin : PluginBase
+{
+	void init(Status status, Attachment attachment, Transaction transaction);
+
+	ProfilerSession startSession(Status status, Transaction transaction, const string description,
+		ISC_TIMESTAMP_TZ timestamp);
+
+	void flush(Status status, Transaction transaction);
+}
+
+interface ProfilerSession : Disposable
+{
+	const uint FLAG_BEFORE_EVENTS = 0x1;
+	const uint FLAG_AFTER_EVENTS = 0x2;
+
```

Comment:
What is the purpose of these flags ?

Reply:
The plugin is able to receive "before" or "after" events, but it may not be interested in both.

The default plugin only deals with "after" events.

"Before" events may be used by some plugin that tracks what code is running and visually show it to the user.